src/components/atoms/Notification.tsx

```diff
@@ -15,6 +15,22 @@ declare module "notistack" {
   }
 }
 
+type Stringifiable = {
+  toString: () => string;
+};
+
+function hasToStringMethod(obj: unknown): obj is Stringifiable {
+  return (
+    typeof obj === "object" &&
+    obj !== null &&
+    typeof (obj as Stringifiable).toString === "function"
+  );
+}
+
+function toString(obj: unknown): string {
+  return hasToStringMethod(obj) ? obj.toString() : "";
+}
+
```

Comment:
With this change we only toString if it's an object (ReactNode in that case) while we actually only want to print when it's a string